src/agoradatatools/process.py

```diff
@@ -159,11 +159,13 @@ def process_dataset(
         entities_as_df[entity_name] = df
 
     if "custom_transformations" in dataset_obj[dataset_name].keys():
-        df = apply_custom_transformations(
+        result = apply_custom_transformations(
             datasets=entities_as_df,
             dataset_name=dataset_name,
             dataset_obj=dataset_obj[dataset_name],
         )
+        df = utils.convert_transformation_result_to_dataframe(result, dataset_name)
```

Comment:
This is a great solution, I'll defer to the rest of the agora team about the output the apply custom transform function expects.

In particular
* Dict
* List of dict
* Dataframe

Do we ever expect just a list of values (non dict) or anything else? If so, that can be another ticket to explore.

Reply:
I don't see a case where we'd format something as a list of values instead of a dict, so I think Dict, list(Dict), and DataFrame are probably sufficient. @beatrizsaldana does that sound right for your Model-AD transforms?